src/engine/OptionalJoin.cpp

```diff
@@ -472,28 +459,20 @@ void OptionalJoin::optionalJoin(const IdTable& dynA, const IdTable& dynB,
       }
     }
   }
-
```

Comment:
In the code above you could also replace `==x.size()` by `>=x.size` to be consistent.